### PR TITLE
feat-661: Add support for build time pass env vars

### PR DESCRIPTION
**Pull request by nightfury1204**
### What is the feature/fix?
https://github.com/convox/issues-private/issues/661

### Add screenshot or video (optional)

https://user-images.githubusercontent.com/12232198/201161466-63e105ec-b0a4-49ea-b1b4-7abe20b03dc9.mp4

### Does it has a breaking change?
no

### How to use/test it?
- deploy or update rack to this branch version
- take the cli of this version
- deploy your app: `convox deploy -a nodejs --build-args MY_TEST=deploy`
- or build your app: `convox build -a nodejs -d cibuild --id --build-args "MY_TEST=build"`
### Checklist
- [ ] New coverage tests
- [x] Unit tests passing
- [x] E2E tests passing
- [x] E2E downgrade/update test passing
- [x] Documentation updated
- [ ] No warnings or errors on Deepsource/Codecov

### CODE REVIEW

**Comment by nightfury1204** [comment]
e2e tests: https://github.com/convox/convox/actions/runs/3438152837/jobs/5734278850

File: cmd/build/main.go [hunk 1]
```diff
@@ -4,15 +4,28 @@ import (
...
 	"github.com/convox/convox/sdk"
 )
 
+type StringSlice []string
+
+func (i *StringSlice) String() string {
+	return strings.Join(*i, ",")
+}
+
+func (i *StringSlice) Set(value string) error {
```
Comment (heronrs) [inline cmd/build/main.go:20]: I guess this is needed to satisfy an interface ?

**Review by heronrs — DISMISSED** [review]
LGTM

File: cmd/build/main.go (hunk 1 above)
Comment (nightfury1204) [reply cmd/build/main.go:20]: yes

**Review by yuriadams — DISMISSED** [review]

**Review by yuriadams — APPROVED** [review]

**Pull request merged**